dora/core/client/fs/src/main/java/alluxio/client/file/dora/DoraCacheClient.java

```diff
@@ -388,14 +391,33 @@ public WorkerNetAddress getWorkerNetAddress(String path) {
       List<BlockWorkerInfo> workers = mContext.getCachedWorkers();
       List<BlockWorkerInfo> preferredWorkers =
           mWorkerLocationPolicy.getPreferredWorkers(workers,
-              path, PREFERRED_WORKER_COUNT);
+              path, mPreferredWorkerCount);
       checkState(preferredWorkers.size() > 0);
-      WorkerNetAddress workerNetAddress = preferredWorkers.get(0).getNetAddress();
+      WorkerNetAddress workerNetAddress = choosePreferredWorker(preferredWorkers).getNetAddress();
       return workerNetAddress;
     } catch (IOException e) {
       // If failed to find workers in the cluster or failed to find the specified number of
       // workers, throw an exception to the application
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Chooses a client preferred worker from multiple workers which hold multiple replicas.
+   *
+   * @param workers a list of workers
+   * @return the preferred worker
+   */
+  protected BlockWorkerInfo choosePreferredWorker(List<BlockWorkerInfo> workers) {
+    return workers.get(0);
+  }
+
+  /**
+   * Get Context.
+   *
+   * @return a file system context
+   */
+  public FileSystemContext getContext() {
+    return mContext;
+  }
```

Comment:
normally the one who created `DoraCacheClient` should have a reference to `FileSystemContext`, so you don't need to expose it via a public getter method. How is this method necessary?

Reply:
This class will be extended in other places. The extended class needs to access the context and then access the mID of the context.